cdci_data_analysis/analysis/email_helper.py

```diff
@@ -416,35 +481,82 @@ def is_email_to_send_callback(logger, status, time_original_request, config, job
                 email_sending_timeout = config.email_sending_timeout
+                info_parameter = 'extracted from the configuration'
 
+            log_additional_info_obj['email_sending_timeout'] = f'{email_sending_timeout}, {info_parameter}'
             logger.info("email_sending_timeout: %s", email_sending_timeout)
+
             logger.info("duration_query > timeout_threshold_email %s", duration_query > timeout_threshold_email)
             logger.info("email_sending_timeout and duration_query > timeout_threshold_email %s",
                         email_sending_timeout and duration_query > timeout_threshold_email)
 
             done_email_files = glob.glob(f'scratch_*_jid_{job_id}*/email_history/*_done_*')
+            log_additional_info_obj['done_email_files'] = done_email_files
             if len(done_email_files) >= 1:
                 logger.info("number of done emails sent: %s", len(done_email_files))
+                log_additional_info_obj['check_result_message'] = 'multiple completion email detected'
+                log_email_sending_info(logger=logger,
+                                       status=status,
+                                       time_request=time_check,
+                                       scratch_dir=scratch_dir,
+                                       job_id=job_id,
+                                       additional_info_obj=log_additional_info_obj
+                                       )
                 raise MultipleDoneEmail("multiple completion email detected")
 
-            return tokenHelper.get_token_user_done_email(decoded_token) and email_sending_timeout and \
+            sending_ok = tokenHelper.get_token_user_done_email(decoded_token) and email_sending_timeout and \
                    duration_query > timeout_threshold_email
+            if sending_ok:
+                email_sending_check_result_msg = 'the email can be sent'
+            else:
+                email_sending_check_result_msg = 'the email cannot be sent'
+            if 'check_result_message' not in log_additional_info_obj:
+                log_additional_info_obj['check_result_message'] = email_sending_check_result_msg
+            else:
+                log_additional_info_obj['check_result_message'] += email_sending_check_result_msg
 
         # or if failed
         elif status == 'failed':
-            return tokenHelper.get_token_user_fail_email(decoded_token)
-
-    return False
+            email_sending_failed = tokenHelper.get_token_user_fail_email(decoded_token)
+            log_additional_info_obj['email_sending_failed'] = email_sending_failed
+            if email_sending_failed:
+                log_additional_info_obj['check_result_message'] = 'the email can be sent'
+            else:
+                log_additional_info_obj['check_result_message'] = 'the email cannot be sent'
+            sending_ok = email_sending_failed
+
+    log_email_sending_info(logger=logger,
```

Comment:
this will happen with every callback, right? This is guaratneed to make the service inoperable, it's often 10 req/second for hours.

Let's make a short log message (no writing files!) in all cases but done/failed/submitted.
Also, better to create new file for every case when the record is actually written (these should be kept few), it's safer.

Reply:
> Let's make a short log message (no writing files!) in all cases but done/failed/submitted.

So basically when the email cannot be sent?

Reply:
when it **is** sent - write file with explanations why was it sent. that's the problem we face - too many emails.
For callback only "done" should be lead to sending. When callback returns failed/submitted - maybe write too, it does not happen often.

write a reasonably short log message when email is not sent with explanation why.

Reply:
Now record is written on a file when email is actually sent, and some explanation on why an email is not sent has been added.